.github/workflows/build_test.yml

```diff
@@ -0,0 +1,30 @@
+name: Java Build, Test
+
+on:
+  push:
+    branches: [ "2.0.0-X" ]
+  pull_request:
+    branches: [ "2.0.0-X" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
```

Comment:
First time I see this, any preference over using the standard `@v4`?

Reply:
Fix the version on its commit hash. @4 is Always the latest of 4
Makes a reproducable build. 
Is recommend from some open source Security observers.

Dependabot can handle the Updates also.

Reply:
Got it, fine by me. Feel free to have a look at the rest of the comments and we can get this merged :+1: